lib/src/visitor/serialize.dart

```diff
@@ -1078,30 +1080,47 @@ class _SerializeVisitor
       _for(value, () => _buffer.write(value.value));
 
   /// Emits [children] in a block.
-  void _visitChildren(List<CssNode> children) {
+  void _visitChildren(Iterable<CssNode> children, CssNode parent) {
     _buffer.writeCharCode($lbrace);
     if (children.every(_isInvisible)) {
       _buffer.writeCharCode($rbrace);
       return;
     }
 
-    _writeLineFeed();
+    bool indentNextChild;
+    if (children.isNotEmpty && _isTrailingComment(children.first, parent)) {
+      _writeOptionalSpace();
+      indentNextChild = false;
+    } else {
+      _writeLineFeed();
+      indentNextChild = true;
+    }
```

Comment:
No, I don't think I forgot to push anything.  The [newer logic](https://github.com/sass/dart-sass/commit/d937f878dabf46abe490c60b3fd4e3671a6ce299#diff-872a211685a7abf16c5f7d36c967296cR1091) should (a) avoid indenting at all if no indentation is necessary (vs previously just indenting one level less) and (b) special-case trailing comments.

I did also try using your suggested code but I wasn't able to make it easily work with the rest of the existing logic in the function (esp the bits around handling semicolons and newlines that follow those).  Unfortunately it's been a little while since I tried so I don't remember all the details.

Reply:
I took another pass at simplifying _visitChildren and managed to get rid of the indentNextChild var.  I'm guessing this is more similar to what you were originally thinking (?).  PTAL.